pkg/controller/clusterpool/clusterpool_controller.go

```diff
@@ -383,6 +377,19 @@ func (r *ReconcileClusterPool) Reconcile(ctx context.Context, request reconcile.
 			log.WithError(err).Error("error adding clusters")
 			return reconcile.Result{}, err
 		}
+	// Delete broken CDs. We put this case after the "addClusters" case because we would rather
+	// consume our maxConcurrent with additions than deletions. But we put it before the
+	// "deleteExcessClusters" case because we would rather trim broken clusters than viable ones.
+	case len(cds.Broken()) > 0:
+		if err := r.deleteBrokenClusters(cds, availableCurrent, logger); err != nil {
+			return reconcile.Result{}, err
+		}
+	// If too many, delete some.
+	case drift > 0:
+		toDel := minIntVarible(drift, availableCurrent)
+		if err := r.deleteExcessClusters(cds, toDel, logger); err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
Note to reviewers: this block moved down unchanged. Since `drift > 0` and `drift < 0` are disjoint, this didn't affect existing logic; but it was necessary to be able to inject the new case in the right spot logically.